internal/repository/repository.go

```diff
@@ -225,13 +225,10 @@ func (r *Repository) SaveAndEncrypt(ctx context.Context, t restic.BlobType, data
 
 	debug.Log("save id %v (%v, %d bytes)", id, t, len(data))
 
-	// get buf from the pool
-	ciphertext := getBuf()
-
-	ciphertext = ciphertext[:0]
 	nonce := crypto.NewRandomNonce()
+
+	ciphertext := make([]byte, 0, len(nonce)+len(data)+r.key.Overhead())
```

Comment:
How about this:
```suggestion
	ciphertext := make([]byte, 0, restic.CiphertextLength(len(data)))
```

Reply:
Force-pushed a new version with this change.